subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/metadata/DefaultArtifactMetadataSource.java

```diff
@@ -61,12 +60,13 @@ public MutableModuleComponentResolveMetadata create(String repositoryName, Compo
         if (metaDataFromDefaultArtifact != null) {
             LOGGER.debug("Found artifact but no meta-data for module '{}' in repository '{}', using default meta-data.", moduleComponentIdentifier, repositoryName);
             metaDataFromDefaultArtifact.getSources()
-                .add(new ModuleDescriptorHashModuleSource(MISSING, false));
+                .add(new ModuleDescriptorHashModuleSource(Hashing.md5().hashString(moduleComponentIdentifier.toString()), false));
```

Comment:
@melix My understanding is that this change does not impact `CacheLayout`. Agreed?

Reply:
It doesn't, indeed. Would be worth extracting the `Hashing` chain into its own method for readability.